packages/esm-ward-app/src/ward-patient-card/ward-patient-card-row.resources.tsx

```diff
@@ -1,100 +1,75 @@
   } = useWardLocation();
-  const { wardPatientCards } = useConfig<WardConfigObject>();
-  const patientCardRows = useMemo(() => {
-    const { cardDefinitions, patientCardElementDefinitions } = wardPatientCards;
 
-    // map of patientCardElementId to its corresponding React component
-    const patientCardElementsMap = new Map<string, WardPatientCardElement>();
-    for (const elementType of builtInPatientCardElements) {
-      patientCardElementsMap.set(
-        elementType,
-        getPatientCardElementFromDefinition({
-          id: elementType,
-          elementType,
-          config: defaultPatientCardElementConfig,
-        }),
+  const currentWardCardConfig = useMemo(() => {
+    const cardDefinition = wardPatientCards.cardDefinitions.find((cardDef) => {
+      return (
+        cardDef.appliedTo == null ||
+        cardDef.appliedTo?.length == 0 ||
+        cardDef.appliedTo.some((criteria) => criteria.location == locationUuid)
       );
-    }
-    for (const patientCardElementDef of patientCardElementDefinitions) {
-      patientCardElementsMap.set(patientCardElementDef.id, getPatientCardElementFromDefinition(patientCardElementDef));
-    }
-
-    const cardDefinition = cardDefinitions.find((cardDef) => {
-      const appliedTo = cardDef.appliedTo;
-
-      return appliedTo == null || appliedTo.some((criteria) => criteria.location == locationUuid);
     });
 
-    const ret = cardDefinition.rows.map((row) => {
-      const { rowType, elements } = row;
-      const patientCardElements = elements.map((patientCardElementId) => {
-        const slot = patientCardElementsMap.get(patientCardElementId);
-        return slot;
-      });
-
-      const WardPatientCardRow: React.FC<WardPatientCardProps> = (props) => {
-        return (
-          <div className={styles.wardPatientCardRow + ' ' + (rowType == 'header' ? styles.wardPatientCardHeader : '')}>
-            {patientCardElements.map((PatientCardElement, i) => (
-              <PatientCardElement {...props} key={i} />
-            ))}
-          </div>
-        );
-      };
+    return cardDefinition;
+  }, [wardPatientCards, locationUuid]);
 
-      return WardPatientCardRow;
-    });
-    return ret;
-  }, [location, wardPatientCards]);
+  if (!currentWardCardConfig) {
+    console.warn(
+      'No ward card configuration has `appliedTo` criteria that matches the current location. Using the default configuration.',
+    );
+    return defaultWardPatientCard;
+  }
 
-  return patientCardRows;
+  return currentWardCardConfig;
 }
 
-function getPatientCardElementFromDefinition(
-  patientCardElementDef: PatientCardElementDefinition,
-): WardPatientCardElement {
-  const { elementType, config } = patientCardElementDef;
-  switch (elementType) {
-    case 'bed-number':
-      return WardPatientBedNumber;
-    case 'patient-name':
-      return WardPatientName;
+export const WardPatientCardElement: React.FC<{ elementId: string; patient: Patient; visit: Visit }> = ({
```

Comment:
Minor, but something which renders a React element, as opposed to data-fetching, feels like it should be in a `.component.tsx` file rather than a `.resource.tsx` file (even though this MFE doesn't do a great job of following the [Coding Conventions](https://openmrs.atlassian.net/wiki/spaces/projects/pages/151061068/Coding+Conventions)).

Reply:
Yeah certainly, lots of files will get renamed after the dust settles a little, before I merge this PR.